### PR TITLE
rpc: warn that nodes ignore requests for old stale blocks

### PR DESCRIPTION
**Pull request by Sjors**
Adds warning to RPC help that `getblockfrompeer` is of little use for stale blocks that are more than a month old.

This is an anti-fingerprinting measure. See `BlockRequestAllowed` in `net_processing`.

It's been in Bitcoin Core since 2014, introduced in #2910 and later improved to not rely on checkpoints.
Older and alternative clients might still serve these blocks, so not throwing an error.

Allowing whitelisted nodes to fetch these blocks anyway might be nice.

### CODE REVIEW

**Comment by Sjors** [comment]
(I noticed this when reviewing #24178)

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -792,7 +792,8 @@ static RPCHelpMan getblockfrompeer()
         "getblockfrompeer",
         "Attempt to fetch block from a given peer.\n\n"
         "We must have the header for this block, e.g. using submitheader.\n"
-        "Subsequent calls for the same block and a new peer will cause the response from the previous peer to be ignored.\n\n"
+        "Subsequent calls for the same block and a new peer will cause the response from the previous peer to be ignored.\n"
+        "Nodes generally ignore requests for a stale block that they never fully verified, or one that is more than a month old.\n\n"
```
Comment (MarcoFalke) [inline src/rpc/blockchain.cpp:796]: This will also disconnect the peer, no? https://github.com/bitcoin/bitcoin/pull/20295#discussion_r764737276

If so, I still think it should be mentioned.

Comment (MarcoFalke) [inline src/rpc/blockchain.cpp:796]:
```suggestion
        "Peers generally ignore requests for a stale block that they never fully verified, or one that is more than a month old.\n\n"
```

Comment (MarcoFalke) [reply src/rpc/blockchain.cpp]: Yes, when I checked at the time, we'd disconnect. But I haven't tested or re-checked since then.

Comment (Sjors) [reply src/rpc/blockchain.cpp]: Afaik there's only one place where we track which blocks we requested: `mapBlocksInFlight`. And afaik that map doesn't get vacuum cleaned. So how would a non-response from a peer lead us to disconnect?

Disclaimer: net_processing is very hard to follow, so I probably missed something.

Comment (MarcoFalke) [reply src/rpc/blockchain.cpp]:
```
                LogPrintf("Timeout downloading block %s from peer=%d, disconnecting\n", queuedBlock.pindex->GetBlockHash().ToString(), pto->GetId());

Comment (Sjors) [reply src/rpc/blockchain.cpp]: Ah yes, we're also updating `state->vBlocksInFlight`, and that does get checked. I'll take a closer look.

**Comment by laanwj** [comment]
I'm not sure about the state of this PR. What are the plans here? Is this a (sufficient) improvement to the documentation, or do you want to do something else instead.

**Comment by Sjors** [comment]
This fell of my radar. I still need to investigate this: https://github.com/bitcoin/bitcoin/pull/24226#discussion_r797446777

Could use some hints from p2p gurus like @jnewbery. The "Waiting for author" tag is fine for now.

**Comment by ajtowns** [comment]
Maybe get the doc improved, and open an issue for the disconnection and fix that later?

**Comment by Sjors** [comment]
I added the disconnect behavior to the RPC doc. Also added a more detailed explanation in `net_processing.cpp`.

File: src/rpc/blockchain.cpp (hunk 1 above)
Comment (Sjors) [reply src/rpc/blockchain.cpp]: Took a closer look. It indeed appears this would disconnect. Updated the doc, but haven't tested it..

**Comment by Sjors** [comment]
Rebased just in case.

**Comment by Sjors** [comment]
@fjahr @luke-jr can you take a look?

**Comment by fjahr** [comment]
Code review ACK f39d9269ebbcffe70d02674c67c4f53783b63005

The improvement of the docs make sense and I agree that, if there is a way to solve this issue, it should happen in a follow-up since it will probably require more discussion.

CI failure looks unrelated, maybe re-run the task @Sjors ?
